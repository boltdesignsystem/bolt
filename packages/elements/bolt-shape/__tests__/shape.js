import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../shape.schema';
// eslint-disable-next-line camelcase
const { border_radius, spacing, size } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
}, 100);

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    content: 'This is a shape',
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt Shape', () => {
  test(`default`, async () => {
    const results = await render('@bolt-elements-shape/shape.twig', {
      ...fixtures.defaultData,
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

describe('Bolt Shape Prop -', () => {
  border_radius.enum.forEach(async option => {
    test(`border_radius items: ${option}`, async () => {
      const results = await render('@bolt-elements-shape/shape.twig', {
        ...fixtures.defaultData,
        border_radius: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  spacing.enum.forEach(async option => {
    test(`spacing: ${option}`, async () => {
      const results = await render('@bolt-components-banner/banner.twig', {
        ...fixtures.defaultData,
        spacing: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  size.enum.forEach(async option => {
    test(`size: ${option}`, async () => {
      const results = await render('@bolt-components-banner/banner.twig', {
        ...fixtures.defaultData,
        size: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
});
