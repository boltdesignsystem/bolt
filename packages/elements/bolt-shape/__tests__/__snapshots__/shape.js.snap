// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Shape Prop - border_radius items: full 1`] = `
<span class="e-bolt-shape e-bolt-shape--border_radius--full">
  <span class="e-bolt-shape__inner">
    This is a shape
  </span>
</span>
`;

exports[`Bolt Shape Prop - border_radius items: large 1`] = `
<span class="e-bolt-shape e-bolt-shape--border_radius--large">
  <span class="e-bolt-shape__inner">
    This is a shape
  </span>
</span>
`;

exports[`Bolt Shape Prop - border_radius items: none 1`] = `
<span class="e-bolt-shape">
  <span class="e-bolt-shape__inner">
    This is a shape
  </span>
</span>
`;

exports[`Bolt Shape Prop - border_radius items: small 1`] = `
<span class="e-bolt-shape e-bolt-shape--border_radius--small">
  <span class="e-bolt-shape__inner">
    This is a shape
  </span>
</span>
`;

exports[`Bolt Shape Prop - size: large 1`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a shape
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt Shape Prop - size: medium 1`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a shape
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt Shape Prop - size: small 1`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a shape
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt Shape Prop - spacing: medium 1`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a shape
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt Shape Prop - spacing: none 1`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a shape
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt Shape Prop - spacing: small 1`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a shape
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt Shape Prop - spacing: xsmall 1`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a shape
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt Shape default 1`] = `
<span class="e-bolt-shape">
  <span class="e-bolt-shape__inner">
    This is a shape
  </span>
</span>
`;
